### PR TITLE
Update statsd-client to export config types use by the main StatsdClient export.

### PR DESCRIPTION
**Pull request by jcass8695**
Update statsd-client to export config types use by the main StatsdClient export.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: The current source code for statsd-client types does not expose the configuration item types. This means in order to extend or wrap the default export `StatsdClient`, we would need to reimplement the config item types or use `any`. [Here](https://gist.github.com/JCass45/249e7c9c85a57ca9455a84d2aff6b4d7) is a gist with an example.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by JCass45** [comment]
@chrisleck If you have time, could I ask you to review this change? You are marked as the owner of these type definitions.

File: types/statsd-client/tsconfig.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
             "es6",
             "dom"
         ],
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/statsd-client/tsconfig.json:8]: AFAIK your package does not require this, that's standard CommonJS module:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44595#discussion_r425400276
https://github.com/msiebuhr/node-statsd-client/blob/master/lib/statsd-client.js#L186
That's for end-users (consumers of the DT module), not  intended to be used here for tests.
So your tests would have to be corrected IMO:
```ts
import SDC = require('statsd-client');
import { Tags, TcpOptions, UdpOptions, HttpOptions } from 'statsd-client';
```

**Comment by orta** [comment]
There's still some unaddressed feedback in this PR

**Comment by orta** [comment]
Ping - The next DT maintainer to check this our should probably close this PR as abandoned if there's no more activity.

**Comment by JCass45** [comment]
@orta I've addressed the outstanding comments

**Review by peterblazejewicz — APPROVED** [review]

**Comment by orta** [comment]
ace, let's get this in then

**Comment by JCass45** [comment]
Thanks folks 👋🏻

**Pull request merged**